Changes for JOSS reviewing process

michael-petersen opened the pull request:
There are several adjustments and improvements to make as part of the JOSS review process; this branch will handle them in one place.

michael-petersen commented:
First, a quality-of-life improvement with adding an issue template.

michael-petersen commented:
Some draft paper updates to resolve #97. 

Anything else we should add, @The9Cat?

The9Cat commented on CMakeLists.txt:
I wonder if we want: 

find_package(Eigen3 3.4...<3.5 REQUIRED)

michael-petersen replied:
I think this is a good improvement; we might want to revisit when Eigen has new releases, but that is sort of inevitable.

The9Cat replied:
The idea here, in part, was to force us (me?) to deal with this when the next Eigen minor becomes stable.

The9Cat commented on INSTALL.md:
I wonder if we need to explain `make -j 4`?   E.g. "The `-j 4` flag allows up to `4` processes to compile simulateously.   Change `4` to the number of cores you can allot to compiling."

michael-petersen commented:
To keep notes in the same place: https://github.com/EXP-code/EXP/issues/95 is being addressed in the relevant repositories, see links in that issue.

michael-petersen commented:
Following up on https://github.com/openjournals/joss-reviews/issues/7302#issuecomment-2430319224:

We've updated and run through the tutorials using the latest main branch of EXP (which is also the Docker image), and think we've caught all the bugs now!

michael-petersen commented on CONTRIBUTING.md:
```diff
@@ -3,7 +3,7 @@
 
 -   Blog about EXP. Cite the EXP published papers (using 
-    the papers in `CITATIONS.bib`). Tell the world how
+    the papers in [`CITATIONS.bib`](https://github.com/EXP-code/EXP/blob/main/CITATIONS.bib)). Tell the world how
```
Note that this link won't work until merge.

michael-petersen commented:
Addressing https://github.com/openjournals/joss-reviews/issues/7302#issuecomment-2430343480:

We've added a CITATIONS.bib; this will also update upon JOSS acceptance to include the citation!

michael-petersen commented:
I believe the last un-addressed comment is https://github.com/openjournals/joss-reviews/issues/7302#issuecomment-2430338844:

1. Do we want to update some images to .svg?
2. I'm imagining the inputs/outputs graphic would be command-line screenshots (or equivalently put into markdown)? Or is there an obvious flowchart idea I'm missing?

The9Cat commented:
As I understand it, the idea is to improve the tutorial walk through showing both code input & output.  It's a good idea but one that would take a bit of time.  My take on the comment is the suggestion to save screenshots from the browser running Jupyter to svg.    There is a Firefox extension for that, I believe.  Random bitmapped graphics won't be improved by svg.

michael-petersen commented:
Many of the `pyEXP-examples` entries have their own notebooks that have input/output stored -- do you think this is enough? Or is the relevant thing to consider what the `EXP-examples` needs for input/output examples?

michael-petersen commented:
@The9Cat has helpfully set up a notebook renderer in https://github.com/EXP-code/EXP-docs/pull/33. This means that for readthedocs, notebooks that are generated as part of the tutorials in [`pyEXP-examples`](https://github.com/EXP-code/pyEXP-examples). Check it out by clicking any of the links in the [Jupyter branch of readthedocs](https://exp-docs.readthedocs.io/en/jupyter/intro/pyEXP-tutorial.html)!

We'll merge this in to `EXP-docs` once everyone is happy.

This addresses https://github.com/openjournals/joss-reviews/issues/7302#issuecomment-2430338844.

The9Cat commented:
The last commit included a few additional wording fixes, added a note about the data to run the examples,  and added navigation stubs to the imported ipynb files back to the Github pyEXP-examples repo.   Looks fine to me now.

michael-petersen commented:
@The9Cat, we've received word this satisfies the reviewers -- can you take a quick look and approve+merge when ready?

The9Cat commented:
Sure, I'll take a quick look now.  I've already reviewed this casually and didn't find any issues.
